### PR TITLE
gh-127274: Defer nested methods

### PR DESCRIPTION
**Pull request by mpage**
Methods (functions defined in class scope) are likely to be cleaned up by the GC anyway, so deferring nested methods will not prolong their lifetime.

Add a new code flag, `CO_METHOD`, that is set for functions defined in a class scope. Use that when deciding to defer functions.

* Issue: gh-127274

### CODE REVIEW

File: Include/cpython/code.h [hunk 1]
```diff
@@ -199,6 +199,9 @@ struct PyCodeObject _PyCode_DEF(1);
 */
 #define CO_HAS_DOCSTRING 0x4000000
 
+/* A function defined in class scope */
+#define CO_METHOD  0x8000000
```
Comment (colesbury) [inline Include/cpython/code.h:203]: I'm not sure it matters, but you could use a smaller constant like `0x0040` (between `CO_GENERATOR` and `CO_COROUTINE`)

**Review by colesbury — APPROVED** [review]
LGTM

**Comment by JelleZijlstra** [comment]
The other `CO_*` constants are documented and exposed in the inspect module (e.g. https://docs.python.org/3.14/library/inspect.html#inspect.CO_HAS_DOCSTRING); we'll have to do that for this one too.

File: Include/cpython/code.h (hunk 1 above)
Comment (JelleZijlstra) [reply Include/cpython/code.h:203]: Could be risky if that reuses a flag that was used for something else in the past. I think it's safer to use a new value unless we're about to run out of bits.

**Comment by mpage** [comment]
> The other CO_* constants are documented and exposed in the inspect module

Should that be done for `CO_NO_MONITORING_EVENTS` as well (in a separate PR)?

**Comment by JelleZijlstra** [comment]
> Should that be done for `CO_NO_MONITORING_EVENTS` as well (in a separate PR)?

Wasn't aware of that flag, but if we also added that one recently, then yes.

**Comment by mpage** [comment]
MacOS-13 build failure looks like an issue with homebrew that is unrelated to this PR.

**Comment by mpage** [comment]
@JelleZijlstra - would you please have another look?

**Review by JelleZijlstra — APPROVED** [review]
Looks good.

Could also add tests that look directly at the flags of a code object, but flags are documented as internal and an implementation detail and CO_HAS_DOCSTRINGS doesn't have tests either, so I think it's fine not to test this.

**Pull request merged**